addons/account/models/account_payment.py

```diff
@@ -132,6 +134,11 @@ class AccountPayment(models.Model):
         compute='_compute_destination_account_id',
         domain="[('account_type', 'in', ('asset_receivable', 'liability_payable'))]",
         check_company=True)
+    end_to_end_id = fields.Char(
+        string='End to End ID',
+        default=lambda s: uuid.uuid4().hex,
+        help='Unique end-to-end assigned by the initiating party',
+    )
```

Comment:
You probably want to index this. Doesn't seem necessary at first on the statements though.